Started writing tests

stasfilin opened the pull request:
Hi @paulthomson

I started to write tests for the repository,
So I decided to mock main requests endpoints

If it's good for you - let's merge this PR, I will continue with writing more tests, e.g. exception tests, mock all GitHub endpoints what we use


Best Regards,
Stanislav Filin

paulthomson commented:
Hi. Thanks for the contribution. Sounds good. And this looks good. I am happy to merge this, but I will add some CI first.

stasfilin commented:
> Hi. Thanks for the contribution. Sounds good. And this looks good. I am happy to merge this, but I will add some CI first.

Hi, okay :)
I will add more tests today


Best Regards,
Stanislav Filin

paulthomson commented:
OK I just added CI. Please rebase this PR and we should see some check results. Please then also uncomment the last line in `check_all.sh`; this should enable the tests.

stasfilin commented:
Hi @paulthomson 
Just finish integrating test to CI (small fixes for annotations, flake8, pylint)

I added few changes for .flake8 config

```
  # Ignore words from requests, like unprocessable
  SC200

  # Ignore hardcoded tokens for tests
  S106
```

Also, I squashed all commits to one


If all good, we can merge this and I will add more tests


Best Regards,
Stanislav Filin

stasfilin commented:
UPD: Changed commit message

paulthomson commented on .flake8:
```diff
@@ -62,6 +62,12 @@ ignore =
   # Pickle and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue.
   S301
 
+  # Ignore words from requests, like unprocessable
+  SC200
```
The spell checker is annoying. However, I would rather you add those words to whitelist.dic and remove this line. There probably won't be many words to add and I have caught a lot of spelling errors thanks to this (although not so much in this project).

paulthomson commented on .flake8:
```diff
@@ -62,6 +62,12 @@ ignore =
+  # Ignore words from requests, like unprocessable
+  SC200
+
+  # Ignore hardcoded tokens for tests
+  S106
```
This is probably a good one to ignore. You can instead do:

```python
 # noqa: S106
```

on the line where you have a "token", but this is probably fine. No action needed.

paulthomson commented on check_all.sh:
```diff
@@ -32,4 +32,4 @@ pylint github_release_retry github_release_retry_tests
 
 # Uncomment once we have some tests.
-# pytest github_release_retry_tests
+pytest github_release_retry_tests
```
Thanks. Could you also remove the comment?

paulthomson commented on github_release_retry_tests/testcase.py:
```diff
@@ -0,0 +1,32 @@
+class GitHubTestCase:
+    @classmethod
+    def get_fixture(cls, filename: str) -> Any:
+        __location__ = os.path.realpath(
+            os.path.join(os.getcwd(), os.path.dirname(__file__))
+        )
+        fixture = os.path.join(__location__, "fixtures", filename)
```
Is it possible to avoiding using the name `__location__` here because of the `__`? Unless this is a convention I am unaware of.

paulthomson commented on github_release_retry_tests/testcase.py:
```diff
@@ -0,0 +1,32 @@
+            os.path.join(os.getcwd(), os.path.dirname(__file__))
+        )
+        fixture = os.path.join(__location__, "fixtures", filename)
+        with open(fixture) as json_file:
```
I tend to prefer using `open(fixture, encoding="utf-8", errors="ignore")` but I am unsure if this is always the best practice.

stasfilin commented:
Hi @paulthomson, thanks for your comments, I push changes with fixes
